src/v/cluster/topics_frontend.cc

```diff
@@ -498,4 +503,76 @@ ss::future<result<model::offset>> topics_frontend::stm_linearizable_barrier(
     });
 }
 
+ss::future<std::vector<topic_result>> topics_frontend::create_partitions(
+  std::vector<create_partititions_configuration> partitions,
+  model::timeout_clock::time_point timeout) {
+    auto r = co_await stm_linearizable_barrier(timeout);
+    if (!r) {
+        std::vector<topic_result> results;
+        results.reserve(partitions.size());
+        std::transform(
+          partitions.begin(),
+          partitions.end(),
+          std::back_inserter(results),
+          [err = r.error()](const create_partititions_configuration& cfg) {
+              return make_error_result(cfg.tp_ns, err);
+          });
+    }
+
+    auto result = co_await ssx::parallel_transform(
+      partitions.begin(),
+      partitions.end(),
+      [this, timeout](create_partititions_configuration cfg) {
+          return do_create_partition(std::move(cfg), timeout);
+      });
+
+    co_return result;
+}
+
+allocation_request make_allocation_request(
+  int16_t replication_factor, const create_partititions_configuration& cfg) {
+    allocation_request req;
+    req.partitions.reserve(cfg.partition_count);
+    for (auto p = 0; p < cfg.partition_count; ++p) {
+        req.partitions.emplace_back(model::partition_id(p), replication_factor);
```

Comment:
are these newly created partitions? why do the ids start at 0 rather than whatever the next id is based on the existing partitions in a topic?

Reply:
I adjust partition Id later,